zone/doors.cpp

```diff
@@ -311,37 +509,107 @@ void Doors::HandleClick(Client* sender, uint8 trigger, bool floor_port)
+						RuleI(Quarm, AutomatedRaidRotationRaidNonMemberCountRequirement),
+						RuleI(Quarm, AutomatedRaidRotationRaidGuildLevelRequirement));
+					return;
+				}
+
+				int64 zonelockout = RuleI(Quarm, InstanceMinimumLockoutTime);
+
+				int64 curzoneLockout = zone->GetZoneInstanceLockoutByCharacterAndZone(sender->CharacterID(), zoneid);
+				auto cur_time = time(nullptr);
+				int64 end_time = cur_time + RuleI(Quarm, InstanceMinimumLockoutTime);
+				if (curzoneLockout <= end_time)
+				{
+					uint32 instanceToDisplay = sender->GetTargetZoneInstanceID(zoneid);
+					sender->Message(Chat::Red, "Please [%s] re-entry to instance #%i. This will re-engage your lockout of %s from %s ( #%i ).", Saylink::Create("#enterzonelockout", true, "confirm").c_str(), instanceToDisplay, Strings::SecondsToTime((int)zonelockout).c_str(), database.GetZoneName(zoneid), instanceToDisplay);
+					sender->SetPendingInstanceDoorID(door_id);
+				}
+				else
+				{
+
+					int64 adjusted_end_time = curzoneLockout - cur_time;
+					uint32 instanceToDisplay = sender->GetTargetZoneInstanceID(zoneid);
+					sender->Message(Chat::Red, "Please [%s] re-entry to instance #%i. This will keep your lockout of %s from %s ( #%i ).", Saylink::Create("#enterzonelockout", true, "confirm").c_str(), instanceToDisplay, Strings::SecondsToTime((int)adjusted_end_time).c_str(), database.GetZoneName(zoneid), instanceToDisplay);
+					sender->SetPendingInstanceDoorID(door_id);
+				}
 				return;
 			}
-
-			if (!player_raid->CanRaidEngageRaidTarget(player_raid->GetLeaderGuildID()))
+			else
 			{
-				sender->Message(Chat::Red, "You are unable to enter a guild instance because you are not a part of a raid containing at least a guild officer as its leader with %i guild members present, and %i players at or above level %i present total.",
-					RuleI(Quarm, AutomatedRaidRotationRaidGuildMemberCountRequirement),
-					RuleI(Quarm, AutomatedRaidRotationRaidNonMemberCountRequirement),
-					RuleI(Quarm, AutomatedRaidRotationRaidGuildLevelRequirement));
-				return;
+				// TODO: GetTargetZoneInstanceID / Get raid leader's instance ID if not leader / if leader and not locked out, issue new ID, if not, message
+				Raid* player_raid = sender->GetRaid();
+
+				if (!player_raid)
+				{
+					sender->Message(Chat::Red, "You are unable to enter an instance because you are not a part of a raid.");
+					return;
+				}
+
+				if (player_raid->MeetsInstancedRaidRequirements() < RuleI(Quarm, AutomatedRaidRotationRaidNonMemberCountRequirement))
+				{
+					sender->Message(Chat::Red, "You are unable to enter an instance because you are not a part of a raid with %i players at or above level %i present total.",
+						RuleI(Quarm, AutomatedRaidRotationRaidNonMemberCountRequirement),
+						RuleI(Quarm, AutomatedRaidRotationRaidGuildLevelRequirement));
+					return;
+				}
+
```

Comment:
same kind of comment as above, there's a good amount of duplicated code in both the if/else that could be moved out for better future maintenance considerations